src/compiler/file_uploader/incomplete_files.ts

```diff
@@ -1,31 +1,37 @@
 import { ActorSubclass, Actor } from '@dfinity/agent';
 import { Src, Dest } from '.';
-import { createAgent } from '../../../dfx';
+import { createAuthenticatedAgent } from '../../../dfx';
 
 export async function getListOfIncompleteFiles(
     paths: [Src, Dest][],
     canisterId: string
 ): Promise<[Src, Dest][]> {
+    const hashActor = await createGetFileHashActor(canisterId);
     const filters = await Promise.all(
-        paths.map(async ([_, destPath]) => {
-            try {
-                let hashActor = await createGetFileHashActor(canisterId);
-                const result = (await hashActor.get_file_hash(destPath)) as
-                    | []
-                    | [string];
-                return result.length === 0;
-            } catch {
-                return true;
-            }
-        })
+        paths.map(
+            async ([_, destPath]): Promise<boolean> =>
+                !(await hasValidHash(destPath, hashActor))
+        )
     );
     return paths.filter((_, index) => filters[index]);
 }
 
+async function hasValidHash(
+    path: string,
+    actor: ActorSubclass
+): Promise<boolean> {
+    try {
+        const hashOption = (await actor.get_file_hash(path)) as [] | [string];
```

Comment:
We shouldn't need to cast these types should we? The actor should be typed appropriately

Reply:
done